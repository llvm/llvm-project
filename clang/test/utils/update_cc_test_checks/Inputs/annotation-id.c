// RUN: %clang_cc1 -ftrivial-auto-var-init=zero -triple=x86_64-unknown-linux-gnu -O1 -disable-llvm-passes -emit-llvm -o - %s | FileCheck %s

void foo_ptr_to_scalar() {
  unsigned long long* a[100];
}
