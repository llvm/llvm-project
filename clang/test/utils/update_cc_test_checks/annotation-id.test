## Test that CHECK lines are generated before the definion and not the declaration

# RUN: cp %S/Inputs/annotation-id.c %t.c && %update_cc_test_checks %t.c
# RUN: diff -u %S/Inputs/annotation-id.expected %t.c
## Check that re-running update_cc_test_checks doesn't change the output
# RUN: %update_cc_test_checks %t.c
# RUN: diff -u %S/Inputs/annotation-id.expected %t.c
