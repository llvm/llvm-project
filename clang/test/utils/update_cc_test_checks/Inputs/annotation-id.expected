// NOTE: Assertions have been autogenerated by utils/update_cc_test_checks.py
// RUN: %clang_cc1 -ftrivial-auto-var-init=zero -triple=x86_64-unknown-linux-gnu -O1 -disable-llvm-passes -emit-llvm -o - %s | FileCheck %s

// CHECK-LABEL: @foo_ptr_to_scalar(
// CHECK-NEXT:  entry:
// CHECK-NEXT:    [[A:%.*]] = alloca [100 x i64*], align 16
// CHECK-NEXT:    [[TMP0:%.*]] = bitcast [100 x i64*]* [[A]] to i8*
// CHECK-NEXT:    call void @llvm.lifetime.start.p0i8(i64 800, i8* [[TMP0]]) #[[ATTR3:[0-9]+]]
// CHECK-NEXT:    [[TMP1:%.*]] = bitcast [100 x i64*]* [[A]] to i8*
// CHECK-NEXT:    call void @llvm.memset.p0i8.i64(i8* align 16 [[TMP1]], i8 0, i64 800, i1 false), !annotation [[ANNT2:![0-9]+]]
// CHECK-NEXT:    [[TMP2:%.*]] = bitcast [100 x i64*]* [[A]] to i8*
// CHECK-NEXT:    call void @llvm.lifetime.end.p0i8(i64 800, i8* [[TMP2]]) #[[ATTR3]]
// CHECK-NEXT:    ret void
//
void foo_ptr_to_scalar() {
  unsigned long long* a[100];
}
